server/api/main.js

```diff
@@ -0,0 +1,30 @@
+import { readFile } from 'fs'
+import { URL } from 'node:url'
+
+let responseCache = null
+
+export default async function handleMain(req, res) {
+  let filePath = new URL(`../../client/dist/index.html`, import.meta.url)
+
+  let sendResponse = content => {
+    res.writeHead(200, {
+      'Content-Type': 'text/html',
+      'Cache-Control': 'public, max-age=300, must-revalidate'
+    })
+    res.end(content, 'utf-8')
+  }
+
+  if (responseCache) {
+    sendResponse(responseCache)
+  } else {
+    readFile(filePath, (error, content) => {
+      if (!error) {
+        sendResponse(content)
+        responseCache = content
```

Comment:
How do we plan to run development version? Will we use Vite server to server HTML? Or will we use the same Node.js server?